### PR TITLE
Add Pimoroni inkyWhat (Red, Black, White) EPD

### PR DESCRIPTION
**Pull request by martinatime**
Got the inkyWhat RBW working as a new screen type. Changed the ploy.py which may affect other screens but I couldn't test as I don't have those.

https://shop.pimoroni.com/products/inky-what?variant=13590497624147

### CODE REVIEW

**Comment by dr-mod** [comment]
Hi Aaron, thank you for your pull request! I'll hopefully have some time tomorrow to take a look at it and get back to you if something pops up.

File: data/plot.py [hunk 1]
```diff
@@ -31,22 +31,22 @@ def center_x(price):
...
-        draw.text((center_x(price), (position_last[1] - position_first[1]) / 2 + position_first[1]), price, font=font)
+        draw.text((center_x(price), (position_last[1] - position_first[1]) / 2 + position_first[1]), price, font=font, fill=fill)
         price = "%d" % min_price
-        draw.text((center_x(price), position_last[1]), price, font=font)
+        draw.text((center_x(price), position_last[1]), price, font=font, fill=fill)
 
     @staticmethod
-    def caption(price, y, screen_width, font, draw):
-        draw.text((-1, y), "BTC", font=font)
+    def caption(price, x, y, screen_width, font, draw, fill=None):
```
Comment (dr-mod) [inline data/plot.py:41]: As the method signature was changed and no default parameter provided the other screens will fail.

File: data/plot.py [hunk 2]
```diff
@@ -31,22 +31,22 @@ def center_x(price):
...
         price_text = "%.2f" % price
         text_width, _ = draw.textsize(price_text, font)
-        price_position = (((screen_width - text_width - 60) / 2) + 60, y)
-        draw.text(price_position, price_text, font=font)
+        price_position = (((screen_width - text_width - x) / 2) + x, y)
+        draw.text(price_position, price_text, font=font, fill=fill)
 
     @staticmethod
-    def candle(data, size=(100, 100), position=(0, 0), draw=None):
+    def candle(data, size=(100, 100), position=(0, 0), draw=None, fill_neg=None, fill_pos=None):
```
Comment (dr-mod) [inline data/plot.py:49]: The fill_neg should probably be "#000000" or black.

Comment (martinatime) [reply data/plot.py:49]: Typically in the financial world black is used for positive while red is used for negative. Also, I didn't want to make assumptions of color values. The inkyWhat seems to have some constants that it uses and I don't know if CSS color names or hex codes would translate across all screens.

File: data/plot.py [hunk 3]
```diff
@@ -31,22 +31,22 @@ def center_x(price):
...
     @staticmethod
-    def caption(price, y, screen_width, font, draw):
-        draw.text((-1, y), "BTC", font=font)
+    def caption(price, x, y, screen_width, font, draw, fill=None):
+        draw.text((x, y), "BTC", font=font, fill=fill)
         price_text = "%.2f" % price
         text_width, _ = draw.textsize(price_text, font)
-        price_position = (((screen_width - text_width - 60) / 2) + 60, y)
-        draw.text(price_position, price_text, font=font)
+        price_position = (((screen_width - text_width - x) / 2) + x, y)
```
[the remark this answers is not in the document]
Comment (martinatime) [reply data/plot.py:45]: I'll see what I can do but as I mentioned in a previous comment Python is not my strongest language

File: README.md [hunk 4]
```diff
@@ -26,20 +26,38 @@ Bitcoin stock price for your RPi Zero
...
-5. To make it run on startup
-    1. `sudo nano /etc/rc.local` 
-    2. Add one the following before `exit 0`
-    ```
-    /usr/bin/python3 /home/pi/zero-btc-screen/main.py &
-    ```
-    conversely, you can run in `screen` you can install it with `sudo apt-get install screen`
-    ```
-    su - pi -c "/usr/bin/screen -dm sh -c '/usr/bin/python3 /home/pi/zero-btc-screen/main.py'"
-    ```
```
[the remark this answers is not in the document]
Comment (martinatime) [reply README.md:42]: ok, I can add that back in

File: configuration.cfg [hunk 5]
```diff
@@ -6,7 +6,8 @@ refresh_interval_minutes : 15
 
 # Enabled screens or devices
 screens : [
-    epd2in13v2
+    inkyWhatRBW
```
[the remark this answers is not in the document]
Comment (martinatime) [reply configuration.cfg:9]: sure, I'll switch it back as the default

File: data/plot.py [hunk 6]
```diff
@@ -31,22 +31,22 @@ def center_x(price):
...
+        draw.text(price_position, price_text, font=font, fill=fill)
```
[the remark this answers is not in the document]
Comment (martinatime) [reply data/plot.py:46]: To make it so it doesn't break existing ones I'll switch it back. I don't know if that is the best default value though

**Comment by dr-mod** [comment]
Hi @martinatime, I've just got around to execute this on my pi and there are two things that I've noticed.
1. I had to install driver for the new screen even though I don't use it. We'll need to think how to resolve thins.
2. After installing the driver it didn't work due to some changes you've made:
```
Traceback (most recent call last):
  File "main.py", line 61, in <module>
    main()
  File "main.py", line 47, in main
    data_sink.update_observers(prices)
  File "/home/pi/zero-btc-screen/presentation/observer.py", line 18, in update_observers
    observer.update(data)
  File "/home/pi/zero-btc-screen/presentation/screens/epd2in13v2.py", line 52, in update
    self.form_image(data, self.screen_draw)
  File "/home/pi/zero-btc-screen/presentation/screens/epd2in13v2.py", line 49, in form_image
    Plot.caption(flatten_prices[len(flatten_prices) - 1], 60, 95, SCREEN_WIDTH, FONT_LARGE, screen_draw)
  File "/home/pi/zero-btc-screen/data/plot.py", line 42, in caption
    draw.text((currency_offset, y), "BTC", font=font, fill=fill)
AttributeError: 'FreeTypeFont' object has no attribute 'text'
```

Will you be able to take a look at the latter please?

**Comment by martinatime** [comment]
I'll look into it this weekend.
I also had to pull in the drivers for the original screens. I'm not certain how you deal with that but the presentation/screens/__init__.py file seems to load all of the screens and therefore needs all of the drivers. My python skills are minimal so I don't know how to programatically load only certain imports in a __init__.py file.
Also, I'll have to deal with the merge conflict introduced by the other PR.

**Comment by dr-mod** [comment]
Great, thanks for looking into it. 
As for having to install drivers for screens you don't need, I've just wrapped imports in "try" this easy fix should resolve this.

**Comment by martinatime** [comment]
I had a little time tonight and it was bugging me. Please review it again and make sure it works on your hardware. Thanks again.

**Comment by dr-mod** [comment]
Hey @martinatime ,  I've come round to double check it, tanks for updating your PR. There is a couple of things that need fixing. Let me do the following, I'll merge the PR now and fix it there, feel free to check if everything works afterwards.

**Pull request merged**